public/app/plugins/datasource/cloudwatch/utils/logsRetry.ts

```diff
@@ -21,16 +21,16 @@ type Result = { frames: DataFrameJSON[]; error?: string };
  * @param options
  */
 export function runWithRetry(
-  queryFun: (targets: StartQueryRequest[]) => Observable<DataFrame[]>,
```

Comment:
before runWithRetry assumed that any errors would be thrown, not inside a response. This meant we called a query function that returned just the frames, assuming the query would throw otherwise. But now we look at the whole response and do the retry logic if there is a specific error in the response